### ISSUE TITLE
refactor(api): Remove `da` API Interface from Celestia Node

### ISSUE THREAD

**Issue by walldiss** [celestiaorg/celestia-node#3932]
### Implementation ideas

# Remove `da` Interface from Celestia Node

---

## What's the Issue?
Right now, our Celestia node includes a `da` (Data Availability) interface from the Rollkit client. It was originally added to fix some dependency issues, but it's not actually part of Celestia’s official API.

## Why Remove It?
- **Extra Work:** We're maintaining something that doesn't need to be here.
- **Confusion:** Users might think `da` is a core part of the Celestia API, which it isn’t.
- **Inconsistency:** Other nodes don’t have this interface, leading to mixed expectations.

## Proposed Solution:
1. **Move the `da` Interface:**
   - **Option A:** Create a separate repository for the `da` interface.
   - **Option B:** Integrate it directly into the Rollkit repository.
2. **Clean Up Celestia Node:**
   - Remove all `da` related code and dependencies.
   - Update the documentation to point users to the new location for `da`.

## Benefits:
- **Less Maintenance:** We can focus on the core features of the Celestia node.
- **Clearer API:** Users will have a better understanding of what’s officially supported.

**Comment by gupadhyaya** [comment]
* are you suggesting we go back to celestia-da? https://github.com/rollkit/celestia-da
* the original issue with celestia-da was that, we had to run celestia-node and this wrapper celestia-da both to eliminate rollkit directly consuming the celestia blob api and having to import it.
* why is it a huge burden to maintain this standalone piece? our original intuition of creating the da interface was that, all da solutions will implement the da interface so that rollkit can communicate directly and doesn't have to know about the da implementation behind the scene.

**Comment by Wondertan** [comment]
So do you say you had to run 3 binaries including Rollkit and having it in node somehow made it 2?

**Comment by gupadhyaya** [comment]
> So do you say you had to run 3 binaries including Rollkit and having it in node somehow made it 2?

yes. this is a major ux improvement for rollups (using rollkit), as 2 binaries is unavoidable (one for rollup and other for da) and bare minimum.

**Comment by walldiss** [comment]
Pls kill it. ev-node was migrated
